corehq/apps/userreports/tests/test_list_expressions.py

```diff
@@ -242,78 +242,103 @@ def test_reduce_items_bad_spec(self, items_ex, reduce_ex):
         'last_item',
         2
     ),
+    (
+        ['joins', 'are', 'cool'],
+        {'type': 'identity'},
+        'join',
+        'joinsarecool',
+    ),
+    # test non-strings
+    (
+        ['joins', 4, 'life'],
+        {'type': 'identity'},
+        'join',
+        'joins4life',
+    ),
+    (
+        [None, 'joins', 'work'],
+        {'type': 'identity'},
+        'join',
+        'joinswork',
+    ),
     # if items can't be compared should return None
     (
         [datetime.datetime.now(), datetime.date.today()],
         {'type': 'identity'},
         'max',
         None,
     ),
-    # if items_expression returns non-iterable reduce(count) should return 0
+
+])
+def test_reduce_items_basic(self, doc, items_ex, reduce_ex, expected):
+    expression = ExpressionFactory.from_spec({
+        'type': 'reduce_items',
+        'items_expression': items_ex,
+        'aggregation_fn': reduce_ex
+    })
+    self.assertEqual(expression(doc), expected)
+
+
+@generate_cases([
     (
-        34,
-        {'type': 'identity'},
```

Comment:
the diff is a little funny here, but just moved these two default arguments into the function, split it out, and then removed comments because the test name is now self-documenting. (same for the empty array tests)